lib/sablon/parser/mail_merge.rb

```diff
@@ -111,13 +111,16 @@ def parse_fields(xml)
       end
 
       private
+
       def build_complex_field(node)
         possible_field_node = node.parent
         field_nodes = [possible_field_node]
         while possible_field_node && possible_field_node.search(".//w:fldChar[@w:fldCharType='end']").empty?
           possible_field_node = possible_field_node.next_element
           field_nodes << possible_field_node
         end
+        # skip instantiation if no end tag
+        return if field_nodes.last.nil?
```

Comment:
What about:
```ruby
ComplexField.new(field_nodes) if field_nodes.last
```
That should be equivalent to your line above unless you dislike one line if statements (I'm only 50/50 on them myself since I come from a Python background)